src/conn/conn_layered.c

```diff
@@ -1460,6 +1474,221 @@ __layered_drain_ingest_tables(WT_SESSION_IMPL *session)
+             * item in the ingest table older than that timestamp must be including in one of the
+             * checkpoints we're saving, and thus can be removed.
+             */
+            ckpt_inuse = layered_table->last_ckpt_inuse;
+            if (ckpt_inuse == 0) {
+                /*
+                 * If we've never checked this layered table before, it's safe to start at the
+                 * oldest checkpoint that we're tracking. There are no cursors in the system that
+                 * are open at checkpoints older than that one. It's probably impossible that we
+                 * haven't tracked any checkpoints, if that happens, we'll start checking at zero.
+                 */
+                if (ds->ckpt_track_cnt > 0)
+                    ckpt_inuse = ds->ckpt_track[ds->ckpt_track_cnt - 1].ckpt_order;
+            }
+
+            /*
+             * Allocate enough room for the uri and the WiredTigerCheckpoint.NNN
+             */
+            len = strlen(layered_table->stable_uri) + strlen(WT_CHECKPOINT) + 20;
+            WT_ERR(__wt_realloc_def(session, &uri_alloc, len, &uri_at_checkpoint));
+
+            /*
+             * For each checkpoint, see of the handle is in use. If not, it is safe to gc.
+             */
+            while (ckpt_inuse < last_ckpt) {
+                WT_ERR(__wt_snprintf(uri_at_checkpoint, uri_alloc, "%s/%s.%" PRId64,
+                  layered_table->stable_uri, WT_CHECKPOINT, ckpt_inuse));
+
+                /* If it's in use, then it must be in the connection cache. */
+                WT_WITH_HANDLE_LIST_READ_LOCK(
+                  session, (ret = __wt_conn_dhandle_find(session, uri_at_checkpoint, NULL)));
+
+                /* If it's in use by any session, then we're done. */
+                if (ret == 0 && session->dhandle->session_inuse > 0)
+                    break;
+
+                WT_ERR_NOTFOUND_OK(ret, false);
+                ++ckpt_inuse;
+            }
+
+            /*
+             * We now have the oldest checkpoint in use for this table. If it's different from the
+             * last time we checked, find out what timestamp that checkpoint corresponds to. That
+             * will be the timestamp we use for pruning.
+             */
+            if (ckpt_inuse != layered_table->last_ckpt_inuse) {
+                for (track = 0; track < ds->ckpt_track_cnt; ++track)
+                    if (ds->ckpt_track[track].ckpt_order == ckpt_inuse)
+                        break;
+                if (track >= ds->ckpt_track_cnt)
+                    WT_ERR_MSG(session, WT_NOTFOUND,
+                      "could not find checkpoint order %" PRId64 " in list of tracked checkpoints",
+                      ckpt_inuse);
+
+                prune_timestamp = ds->ckpt_track[track].timestamp;
+                /*
+                 * TODO: this assumes the ingest table dhandle is open if the layered table dhandle
```

Comment:
This assumption turns to be wrong. The following assert is firing and the btree is garbage. @ddanderson What is the better way to get the btree of the ingest table?